entrypoint.sh

```diff
@@ -1,46 +1,52 @@
 #!/bin/sh
 
-touch ${SSH_KEY_FILE:=/id_rsa}
-chmod 0400 ${SSH_KEY_FILE:=/id_rsa}
+# Set up key file
+KEY_FILE=${SSH_KEY_FILE:=/id_rsa}
+if [ ! -f "${KEY_FILE}" ]; then
+	echo "[ERROR] No SSH Key file found"
+	exit 1
+fi
+eval $(ssh-agent -s)
+cat "${SSH_KEY_FILE}" | ssh-add -k -
 
+# Set up known_hosts file if needed
 STRICT_HOSTS_KEY_CHECKING=no
 KNOWN_HOSTS=${SSH_KNOWN_HOSTS:=/known_hosts}
 if [ -f "${KNOWN_HOSTS}" ]; then
-    chmod 0400 ${KNOWN_HOSTS}
-    KNOWN_HOSTS_ARG="-o UserKnownHostsFile=${KNOWN_HOSTS}"
+    KNOWN_HOSTS_ARG="-o UserKnownHostsFile=${KNOWN_HOSTS} -o CheckHostIP=no "
     STRICT_HOSTS_KEY_CHECKING=yes
 fi
 
 # Pick a random port above 32768
 DEFAULT_PORT=$RANDOM
 let "DEFAULT_PORT += 32768"
-echo [INFO] Tunneling ${SSH_HOSTUSER:=root}@${SSH_HOSTNAME:=localhost}:${SSH_TUNNEL_REMOTE:=${DEFAULT_PORT}} to ${SSH_TUNNEL_HOST=localhost}:${SSH_TUNNEL_LOCAL:=22}
```

Comment:
Yeah, this was just my newbishness of creating dockerfiles showing when I initially created this years ago.  Thanks for fixing it.